spec/query_associations_spec.cr

```diff
@@ -21,4 +21,96 @@ describe "Query associations" do
     posts = Post::BaseQuery.new.join_comments.comments(&.body("matching"))
     posts.results.should eq([post_with_matching_comment])
   end
+
+  it "can query associations with inner_join specified" do
+    post_with_matching_comment = PostBox.create
+    matching_comment = CommentBox
+      .new
+      .body("matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+    post_without_matching_comment = PostBox.create
+    CommentBox
+      .new
+      .body("not-matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+
+    posts = Post::BaseQuery.new.inner_join_comments.comments(&.body.eq("matching"))
+    posts.to_sql[0].should contain "INNER JOIN"
+    posts.results.should eq([post_with_matching_comment])
+
+    posts = Post::BaseQuery.new.inner_join_comments.comments(&.body("matching"))
+    posts.to_sql[0].should contain "INNER JOIN"
+    posts.results.should eq([post_with_matching_comment])
+  end
+
+  it "can query associations with left_join specified" do
+    post_with_matching_comment = PostBox.create
```

Comment:
Also might be good to have another Post called `post_without_matching_comment` that can be used with the second comment. That way we know it is excluded.

```crystal
CommentBox.new
  .body("not-matching")
  .post_id(post_without_matching_comment)
  .create

# Your current assertion should still pass if the join worked right
posts.results.should eq([post_with_matching_comment])
```

Reply:
that's already in the spec though 🤔 All of them have that... Or am I missing something?

Reply:
Right now there are two comments, but just one post. So the second comment isn't really testing much. By creating another post and making the second comment attached to it you can be sure that the join query is working expected and only returning the post related to the matching comment and not the second post. Does that make sense?

Reply:
There's 2 posts though... On all of them. Yeah, I'm still a bit confused.

```crystal
  # first post and comment
    post_with_matching_comment = PostBox.create
    matching_comment = CommentBox
      .new
      .body("matching")
      .post_id(post_with_matching_comment.id)
      .create

  # second post and comment
    post_without_matching_comment = PostBox.create
    CommentBox
      .new
      .body("not-matching")
      .post_id(post_with_matching_comment.id)
      .create
```

Reply:
You are *totally* right. I didn't see the second post, but also the second comment calls `post_with_matching_comment` in every spec which I think is a mistake. The second post is never used in these. Maybe add a newline after the first comment as well so it is easier to see? Also I guess I need glasses 🤣

Reply:
hahahah I missed that too. Ok. I'll update that. 😆